### PR TITLE
Modify EthTx adapter to support multiword

### PR DESCRIPTION
**Pull request by connorwstein**
Note that there is a follow up ticket to write a big integration test for using the result collect + ABIEncoding together.

### CODE REVIEW

File: core/adapters/eth_tx_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
+package adapters
```
Comment (connorwstein) [inline core/adapters/eth_tx_internal_test.go:1]: This is an intentional whitebox test, I think it makes sense for this case and I think we could continue with the pattern: xxx_test.go (blackbox test) xxx_internal_test.go (whitebox test)

File: core/adapters/eth_tx.go [hunk 2]
```diff
@@ -26,9 +31,15 @@ type EthTx struct {
 	FromAddress      common.Address          `json:"fromAddress,omitempty"`
 	FromAddresses    []common.Address        `json:"fromAddresses,omitempty"`
 	FunctionSelector models.FunctionSelector `json:"functionSelector"`
-	DataPrefix       hexutil.Bytes           `json:"dataPrefix"`
-	DataFormat       string                  `json:"format"`
-	GasLimit         uint64                  `json:"gasLimit,omitempty"`
+	// DataPrefix is typically a standard first argument
+	// to chainlink callback calls - usually the requestID
```
Comment (samsondav) [inline core/adapters/eth_tx.go:35]: Good comment

File: core/adapters/eth_tx.go [hunk 3]
```diff
@@ -177,6 +201,101 @@ func getConfirmedTxHash(ethTxID int64, db *gorm.DB, minRequiredOutgoingConfirmat
...
+
+func init() {
+	for k := range solidityTypeToGoType {
+		supportedSolidityTypes = append(supportedSolidityTypes, k)
+	}
+}
+
+// Note we need to include the data prefix handling here because
+// if dynamic types (such as bytes) are used, the offset will be affected.
+func getTxDataUsingABIEncoding(e *EthTx, inputData models.JSON) ([]byte, error) {
```
Comment (samsondav) [inline core/adapters/eth_tx.go:226]: Unit test on this?

File: core/adapters/eth_tx_internal_test.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+			args:        []interface{}{18},
+			errLike:     "uint8 is unsupported", // Could consider relaxing this to just <= 32?
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			j := models.JSON{}
+			d, err := j.Add(models.ResultCollectionKey, tc.args)
+			require.NoError(t, err)
+			b, err := getTxDataUsingABIEncoding(&EthTx{
```
Comment (samsondav) [inline core/adapters/eth_tx_internal_test.go:144]: Ah, here it is

**Review by samsondav — APPROVED** [review]

**Pull request merged**